pkg/api/v1/conversion.go

```diff
@@ -480,10 +480,9 @@ func Convert_v1_PodTemplateSpec_To_api_PodTemplateSpec(in *PodTemplateSpec, out
 		in.Spec.InitContainers = values
 
 		// Call defaulters explicitly until annotations are removed
-		for i := range in.Spec.InitContainers {
-			c := &in.Spec.InitContainers[i]
-			SetDefaults_Container(c)
-		}
+		tmpPod := &Pod{Spec: PodSpec{InitContainers: values}}
+		SetObjectDefaults_Pod(tmpPod)
```

Comment:
I suppose 1.6 kubelets are the first to report initcontainer status in fields, so in 1.8 we can drop the annotations and use only the fields